src/main/java/fi/metatavu/metamind/bot/slots/LocalizedNumberSlot.java

```diff
@@ -0,0 +1,169 @@
+package fi.metatavu.metamind.bot.slots;
+
+import com.rabidgremlin.mutters.core.Context;
+import com.rabidgremlin.mutters.core.SlotMatch;
+import com.rabidgremlin.mutters.slots.NumberSlot;
+
+/**
+ * Class to extend mutters NumberSlot to support multiple languages
+ * 
+ * @author Heikki Kurhinen
+ *
+ */
+public class LocalizedNumberSlot extends NumberSlot {
+
+  private static final String[] SUPPORTED_LANGUAGES = {
+    "finnish",
+    "english"
+  };
+  
+  private static final int MAX_TOKENS = 100;
+  
+  public LocalizedNumberSlot(String name) {
+    super(name);
+  }
+
+  @Override
+  public SlotMatch match(String token, Context context) {
+
+    try {
+      Long value = Long.parseLong(token);
+
+      return new SlotMatch(this, token, value);
+    } catch (NumberFormatException de) {
+      
+      try {
+        Double value = Double.parseDouble(token);
+        
+        return new SlotMatch(this, token, value);
+      } catch (NumberFormatException le) {
+
+        Number value = localizedWordStringToNumber(token);
+        if (value != null) {
+          return new SlotMatch(this, token, value);
+        } else {
+          return null;
+        }
+      }
+    }
+  }
+  
+  private Number localizedWordStringToNumber(String token) {
+    for (String language : SUPPORTED_LANGUAGES) {
+      Number result = null;
+      switch (language) {
+        case "finnish":
+          result = finnishWordStringToNumber(token);
+        break;
+        case "english":
+          result = englishWordStringToNumber(token);
+        break;
+      }
+      
+      if (result != null) {
+        return result;
+      }
+    }
+
+    return null;
+  }
+  
+  private Number englishWordStringToNumber(String token) {
+    return wordStringToNumber(token);
+  }
+  
+  private Number finnishWordStringToNumber(String token) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 25 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)